presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -254,17 +256,19 @@ private InternalTable buildPartitions(Session session, String catalogName, Map<S
                         String columnName = columnHandles.get(columnHandle);
                         String value = null;
                         if (entry.getValue().getValue() != null) {
-                            ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle.get(), columnHandle);
-                            try {
-                                Signature operator = metadata.getFunctionRegistry().getCoercion(columnMetadata.getType(), VARCHAR);
-                                value = ((Slice) metadata.getFunctionRegistry().getScalarFunctionImplementation(operator).getMethodHandle().invokeWithArguments(entry.getValue().getValue())).toStringUtf8();
-                            }
-                            catch (OperatorNotFoundException e) {
-                                value = "<UNREPRESENTABLE VALUE>";
-                            }
-                            catch (Throwable throwable) {
-                                throw Throwables.propagate(throwable);
-                            }
+                            value = values.computeIfAbsent(columnHandle, column -> {
```

Comment:
I think this is going to do too much caching. This caches a single value for each column, but you have multiple domains which each have different values. My suggestion was to cache `Map<Type, MethodHandle>` which caches just the result of
```
Signature operator = metadata.getFunctionRegistry().getCoercion(columnMetadata.getType(), VARCHAR);
metadata.getFunctionRegistry().getScalarFunctionImplementation(operator).getMethodHandle()
```